prusti-encoder/src/encoders/mir_pure.rs

```diff
@@ -445,39 +455,77 @@ impl<'tcx, 'vir: 'enc, 'enc> Encoder<'tcx, 'vir, 'enc>
                 #[derive(Debug)]
                 enum PrustiBuiltin {
                     Forall,
+                    SnapshotEquality,
+                    Pure(DefId),
```

Comment:
Rename the enum or make this a separate mechanism. Ideally the builtin stuff could be extracted and only deal with figuring out if a `DefId` is a Prusti builtin.

Reply:
I've moved it to a separate function for now (and removed the `Pure` variant), we can extract more later